src/Data/Text.hs

```diff
@@ -1760,7 +1781,8 @@ filter p = go
       -- is filtered with a rare predicate, resulting in a much shorter buffer.
       let !dstLen = min l 64
       dst <- A.new dstLen
-      outer dst dstLen o 0
+      Text arr off len <- outer dst dstLen o 0
+      pure $ text arr off len
```

Comment:
Wouldn't it  be better to keep the tail call to `outer`?

Reply:
The reason I do this is to allow `text` to be cancelled out by the `text/chunk` rule when this function is inlined into the Lazy.Text equivalent. If I have `text` be inside `outer` then I don't think the rule matches. I'm not convinced this is the best approach though.

Reply:
Done in 685411e1201558e815c9df5c6f359f60956b007f